mitodl/open-discussions
# Integrate widget framework

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1441 

#### What's this PR do?
Adds the widget framework

#### How should this be manually tested?

- Rebuild the app `docker-compose build`
- Migrate the app `docker-compose up`
- Verify you can view channels and posts without errors
- Run `./manage.py backpopulate_channel_widget_lists` to generate default widgets based off the channel descriptions
- Test modifying the widget list as a moderator user via the api:

API:

- GET `/api/v0/widget-lists/:id/`
- PATCH `/api/v0/widget-lists/:id/`

Example curl modifying the widget list (you'll need to add session info or comment out `permission_classes` on the view):

```
curl --header "Content-Type: application/json" \
     --request PATCH \
     --data '{"widgets":[{"widget_type": "Markdown", "title": "Test Title", "configuration": {"source": "*markdown*"}}]}' \
     http://od.dev.local:8063/api/v0/widget-lists/1/
```
